### ISSUE TITLE
Potential buffer overflows

### ISSUE THREAD

**Issue by valleyboy7** [RetroShare/RetroShare#1444]
During a quick source code review, I noticed a potential problem in the following lines:

https://github.com/RetroShare/RetroShare/blob/master/libretroshare/src/services/p3idservice.cc#L1862
https://github.com/RetroShare/RetroShare/blob/master/libretroshare/src/services/p3idservice.cc#L1869
https://github.com/RetroShare/RetroShare/blob/master/libretroshare/src/services/p3idservice.cc#L2068

In all three cases, it has sscanf() calls with "%[]" format strings, copying into a fixed size buffer. If it's possible in either case for the copied data to be larger than the buffers, this might lead to crashes or even remote code execution.

What about changing "%[" to "%1023[" in the format strings?

**Comment by valleyboy7** [comment]
https://pastebin.com/2B14BXcU

**Comment by G10h4ck** [comment]
Thanks for taking a look at the code, a pull request to fix this is welcome, @csoler what do you think of this report?

**Comment by csoler** [comment]
I agree that is can be  problem. Although service strings are produced locally and never exchanged over the network, so the threat is quite limited. Anyway, because pgpline is of fixed size, limiting the size that is read totally makes sense.
